src/vm/threads.cpp

```diff
@@ -8332,31 +8358,59 @@ LONG Thread::GetTotalThreadPoolCompletionCount()
     }
     CONTRACTL_END;
 
-    LONG total;
+    UINT64 total;
     if (g_fEEStarted) //make sure we actually have a thread store
     {
-        // make sure up-to-date thread-local counts are visible to us
-        ::FlushProcessWriteBuffers();
-
         // enumerate all threads, summing their local counts.
         ThreadStoreLockHolder tsl;
 
-        total = s_threadPoolCompletionCountOverflow.Load();
+        total = GetOverflowCount(overflowCount);
 
         Thread *pThread = NULL;
         while ((pThread = ThreadStore::GetAllThreadList(pThread, 0, 0)) != NULL)
         {
-            total += pThread->m_threadPoolCompletionCount;
+            total += *GetThreadLocalCountRef(pThread, threadLocalCountOffset);
         }
     }
     else
     {
-        total = s_threadPoolCompletionCountOverflow.Load();
+        total = GetWorkerThreadPoolCompletionCountOverflow();
     }
 
     return total;
 }
 
+UINT64 Thread::GetTotalThreadPoolCompletionCount()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    UINT64 total;
+    if (g_fEEStarted) //make sure we actually have a thread store
```

Comment:
This condition looks like a left-over from desktop. I think it is fine to delete it.